src/at-operator-for-array-access/README.md

```diff
@@ -0,0 +1,40 @@
+An "at" operator for array access
+=================================
+
+It can be useful to have a concise notation for accessing a multidmensional
+array by mapping each component of an indicial vector to an index of the
+accessed array sequentially. For example, in cloud microphysics, aerosol
+particles may serve as nucleation sites for droplet formation and it might
+be useful to gather statistics on the relative humidity conditioned on the 
+presence of an aerosol particle for purposes of predicting cloud formation.  
+For a given particle, we might define an 3D, integer vector `nearest_neighbor`
+in which we have stored the indicial values for to the grid cell nearest the
+particle.  Then a humidity can be calculated conditioned upon particle 
+presence by averaging humidities at locations of the following form:
+```fortran
+conditional_humidity = relative_humidity( nearest_neighbor(1),nearest_neighbor(2),nearest_neighbor(3) )
+```
+
+In Mathematica, a much more compact notation exists for such a purpose:
+```fortran
```

Comment:
I think Github knows about Mathematica and you could safely replace

    ```fortran

with

    ```mathematica

Here is a test rendering:
```mathematica
conditional_humidity  = relative_humidity  @@ nearest_neighbor 
```

Reply:
Ideally a similar approach can also be used for array allocation.   Would be straightforward for default lower bounds.  Not sure there is a good substitute for situations where you want to specify upper and lower bounds.